sematic/db/models/job.py

```diff
@@ -0,0 +1,215 @@
+        updated_at relates to updates of the DB record. It is in epoch seconds
+        rather than as a datetime object because it will primarily be used in
+        arithmetic operations with time (ex: comparing which value is more recent,
+        amount of elapsed time since last update) rather than for human-readability
+        of absolute time.
+    created_at:
+        The time this record was created in the DB.
+    updated_at:
+        The time this record was last updated in the DB. See documentation in
+        last_updated_epoch_seconds for how this relates to that field.
+    """
+
+    # Q: Why duplicate data that's already in the json of
+    # status_history_serializations and value_serialization as columns?
+    # A: For two reasons:
+    #    1. It gives us freedom to refactor the dataclass for
+    #    JobStatus/ExternalJob/KubernetesExternalJob later to move fields around,
+    #    while allowing the database columns to stay stable.
+    #    2. It allows for more efficient queries on the explicit columns rather than
+    #    requiring json traversal.
+
+    __tablename__ = "jobs"
+
+    id: str = Column(types.String(), primary_key=True)
+    source_run_id: str = Column(  # type: ignore
+        types.String(),
+        nullable=False,
+    )
+    state_name: str = Column(  # type: ignore
+        types.String(),
+        nullable=False,
+    )
+
+    # Why do we need both state_name and is_active? The former is fairly
+    # free-form and subject to change. The latter should be a stable way
+    # to determine whether the job is in a terminal state or not, and
+    # whether it is expected to change.
+    is_active: int = Column(
+        types.Integer, nullable=False  # sqlite doesn't support bool
+    )
+    status_message: str = Column(types.String(), nullable=False)
+    job_type: JobType = Column(  # type: ignore
+        types.Enum(JobType),
+        nullable=False,
+    )
+    last_updated_epoch_seconds: float = Column(types.Float(), nullable=False)
+    value_serialization: Dict[str, Any] = Column(types.JSON(), nullable=False)
+    status_history_serializations: Tuple[Dict[str, Any], ...] = Column(  # type: ignore
+        types.JSON(), nullable=False
+    )
+    created_at: datetime.datetime = Column(
+        types.DateTime(), nullable=False, default=datetime.datetime.utcnow
+    )
+    updated_at: datetime.datetime = Column(
+        types.DateTime(),
+        nullable=False,
+        default=datetime.datetime.utcnow,
+        onupdate=datetime.datetime.utcnow,
+    )
+
+    @validates("job_type")
+    def validate_job_type(self, key: Any, job_type: Union[str, JobType]) -> JobType:
+        if isinstance(job_type, str):
+            return JobType[job_type]
+        elif isinstance(job_type, JobType):
+            return job_type
+        raise ValueError(f"Cannot make a JobType from {job_type}")
+
+    @classmethod
+    def from_job(cls, job: ExternalJob, run_id: str) -> "Job":
```

Comment:
`from_external_job` for consistency?

Reply:
Additional question, should `ExternalJob` and `Job` be merged?

Reply:
> `from_external_job` for consistency?

Sure

> Additional question, should ExternalJob and Job be merged?

They could be, with a lot of work. It would mean (a) DB migration of existing runs (which store external jobs) and (b) rewriting a bunch of code to work in a "SQLAlchemy model-y way" rather than in a "dataclass-y way". If we're going to go that far, then we should probably just go for the refactor of stopping including external jobs inside the run objects, since it's basically just as hard.